packages/puppeteer-core/src/common/Page.ts

```diff
@@ -146,58 +142,59 @@ export class CDPPage extends Page {
   #serviceWorkerBypassed = false;
   #userDragInterceptionEnabled = false;
 
-  #frameManagerHandlers = new Map<symbol, Handler<any>>([
+  #frameManagerHandlers = Object.freeze([
     [
-      FrameManagerEmittedEvents.FrameAttached,
-      this.emit.bind(this, PageEmittedEvents.FrameAttached),
+      FrameManagerEvent.FrameAttached,
+      this.emit.bind(this, PageEvent.FrameAttached),
     ],
     [
-      FrameManagerEmittedEvents.FrameDetached,
-      this.emit.bind(this, PageEmittedEvents.FrameDetached),
+      FrameManagerEvent.FrameDetached,
+      this.emit.bind(this, PageEvent.FrameDetached),
     ],
     [
-      FrameManagerEmittedEvents.FrameNavigated,
-      this.emit.bind(this, PageEmittedEvents.FrameNavigated),
+      FrameManagerEvent.FrameNavigated,
+      this.emit.bind(this, PageEvent.FrameNavigated),
     ],
-  ]);
+  ] as const);
 
-  #networkManagerHandlers = new Map<symbol, Handler<any>>([
-    [
-      NetworkManagerEmittedEvents.Request,
-      this.emit.bind(this, PageEmittedEvents.Request),
-    ],
-    [
-      NetworkManagerEmittedEvents.RequestServedFromCache,
-      this.emit.bind(this, PageEmittedEvents.RequestServedFromCache),
-    ],
+  #networkManagerHandlers = Object.freeze([
+    [NetworkManagerEvent.Request, this.emit.bind(this, PageEvent.Request)],
     [
-      NetworkManagerEmittedEvents.Response,
-      this.emit.bind(this, PageEmittedEvents.Response),
+      NetworkManagerEvent.RequestServedFromCache,
+      this.emit.bind(this, PageEvent.RequestServedFromCache),
     ],
+    [NetworkManagerEvent.Response, this.emit.bind(this, PageEvent.Response)],
     [
-      NetworkManagerEmittedEvents.RequestFailed,
-      this.emit.bind(this, PageEmittedEvents.RequestFailed),
+      NetworkManagerEvent.RequestFailed,
+      this.emit.bind(this, PageEvent.RequestFailed),
     ],
     [
-      NetworkManagerEmittedEvents.RequestFinished,
-      this.emit.bind(this, PageEmittedEvents.RequestFinished),
+      NetworkManagerEvent.RequestFinished,
+      this.emit.bind(this, PageEvent.RequestFinished),
     ],
-  ]);
+  ] as const);
 
-  #sessionHandlers = new Map<symbol | string, Handler<any>>([
+  #sessionHandlers = Object.freeze([
     [
-      CDPSessionEmittedEvents.Disconnected,
+      CDPSessionEvent.Disconnected,
       () => {
-        return this.#sessionCloseDeferred.resolve(
+        this.#sessionCloseDeferred.resolve(
           new TargetCloseError('Target closed')
         );
       },
     ],
     [
       'Page.domContentEventFired',
-      this.emit.bind(this, PageEmittedEvents.DOMContentLoaded),
+      () => {
+        return this.emit(PageEvent.DOMContentLoaded, undefined);
```

Comment:
Why this change, can't you pass `undefined` as 3th parameter?

Reply:
TypeScript has trouble without the above.